src/Plugins/ComplexCore/src/ComplexCore/Filters/MinNeighbors.cpp

```diff
@@ -0,0 +1,365 @@
+  std::vector<int32> n(numfeatures + 1, 0);
+  while(counter != 0)
+  {
+    counter = 0;
+    for(int64 k = 0; k < dims[2]; k++)
+    {
+      kstride = dims[0] * dims[1] * k;
+      for(int64 j = 0; j < dims[1]; j++)
+      {
+        jstride = dims[0] * j;
+        for(int64 i = 0; i < dims[0]; i++)
+        {
+          count = kstride + jstride + i;
+          featurename = featureIds[count];
+          if(featurename < 0)
+          {
+            counter++;
+            current = 0;
+            most = 0;
+            for(int32 l = 0; l < 6; l++)
+            {
+              good = 1;
+              neighpoint = count + neighpoints[l];
+              if(l == 0 && k == 0)
+              {
+                good = 0;
+              }
+              if(l == 5 && k == (dims[2] - 1))
+              {
+                good = 0;
+              }
+              if(l == 1 && j == 0)
+              {
+                good = 0;
+              }
+              if(l == 4 && j == (dims[1] - 1))
+              {
+                good = 0;
+              }
+              if(l == 2 && i == 0)
+              {
+                good = 0;
+              }
+              if(l == 3 && i == (dims[0] - 1))
+              {
+                good = 0;
+              }
+              if(good == 1)
+              {
+                feature = featureIds[neighpoint];
+                if(feature >= 0)
+                {
+                  n[feature]++;
+                  current = n[feature];
+                  if(current > most)
+                  {
+                    most = current;
+                    neighbors[count] = neighpoint;
+                  }
+                }
+              }
+            }
+            for(int32 l = 0; l < 6; l++)
+            {
+              good = 1;
+              neighpoint = count + neighpoints[l];
+              if(l == 0 && k == 0)
+              {
+                good = 0;
+              }
+              if(l == 5 && k == (dims[2] - 1))
+              {
+                good = 0;
+              }
+              if(l == 1 && j == 0)
+              {
+                good = 0;
+              }
+              if(l == 4 && j == (dims[1] - 1))
+              {
+                good = 0;
+              }
+              if(l == 2 && i == 0)
+              {
+                good = 0;
+              }
+              if(l == 3 && i == (dims[0] - 1))
+              {
+                good = 0;
+              }
+              if(good == 1)
+              {
+                feature = featureIds[neighpoint];
+                if(feature >= 0)
+                {
+                  n[feature] = 0;
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    for(usize j = 0; j < totalPoints; j++)
+    {
+      featurename = featureIds[j];
+      neighbor = neighbors[j];
+      if(featurename < 0 && neighbor >= 0 && featureIds[neighbor] >= 0)
+      {
+        for(const auto& arrayPath : voxelArrayPaths)
```

Comment:
`getData` can potentially be expensive depending on the depth of the path, so the arrays should be acquired before the loops.